### PR TITLE
#1597 copying overrides clipboard

### LINKED ISSUE #1597: Copying operations in different labs should not override clipboard items
For each of the different labs, some of the labs replaces the clipboard image. Would be good to use the ClipboardUtil method that prevents the clipboard from being overridden.
1. Sync Lab
- [x] Copy #1597

2. Effects Lab

- [ ] Magnifying glass
- [x] Create Spotlight #1597
- [x] Blur Selected #1597
- [x] Blur Remainder #1597
- [x] Blur all except selected #1597
- [x] Recolor remainder #1597
- [x] Recolor all except selected #1597

3. Shortcuts Lab

- [x] Convert to picture #1580
- [x] Add to shapes lab #1636 

4. Paste Lab

- [x] Paste into group #1634 

5. Agenda Lab

- [x] Create Text Agenda #1597
- [x] Create Visual Agenda #1597
- [x] Create Sidebar Agenda #1597
- [x] Sync agenda #1597

6. Zoom Lab

- [x] Drill down #1597
- [x] Step back #1597
- [x] Zoom to area #1597

### PR DESCRIPTION
**Pull request by iskandarzulkarnaien**
Fixes #1597 

Please use the following checklist to help check whether the fix works as intended on your version of Office

Effects Lab:
- [ ] Create Spotlight
- [ ] Blur Selected
- [ ] Blur Remainder
- [ ] Blur All Except Selected
- [ ] Recolor Remainder
- [ ] Recolor All Except Selected

Sync Lab
- [ ] Copy

Shortcuts Lab
- [ ] Add to Shapes Lab

Agenda Lab
- [ ] Create Text Agenda
- [ ] Create Visual Agenda
- [ ] Create Sidebar Agenda
- [ ] Sync Agenda

Zoom Lab:
- [ ] Drill Down
- [ ] Step Back
- [ ] Zoom To Area

`ConvertShapesToTooltip` UT Failed (however it also fails on dev-branch for me, can someone test and let me know if the results are similar?)

**Outline of Solution**
Used `ClipboardUtil.RestoreClipboardAfterAction` method as mentioned in #1597.

### CODE REVIEW

File: PowerPointLabs/PowerPointLabs/SyncLab/SyncLabShapeStorage.cs [hunk 1]
```diff
@@ -124,7 +135,7 @@ public string CopyShape(Shape shape, Format[] formats)
             string shapeKey = nextKey.ToString();
             nextKey++;
             copiedShape.Name = shapeKey;
-
+
```
[the remark this answers is not in the document]
Comment (iskandarzulkarnaien) [reply PowerPointLabs/PowerPointLabs/SyncLab/SyncLabShapeStorage.cs]: Removed, thanks!

**Comment by iskandarzulkarnaien** [comment]
I just added a commit that fixes the same issue in the zoom lab

**Comment by iskandarzulkarnaien** [comment]
The proposed fix introduces a bug to zoom lab. Attempting to drilldown/stepback on the last/first slide respectively fails silently instead of informing the user that is an illegal operation

**Comment by iskandarzulkarnaien** [comment]
Ready for review

**Comment by iskandarzulkarnaien** [comment]
Ready for review

**Comment by ccyccyccy** [comment]
Tested for 2016.

Sidenote: Bug still remains where it adds duplicate items to Powerpoint's clipboard pane as well as Window's clipboard history, but unsure if those can be fixed. Related: https://github.com/PowerPointLabs/PowerPointLabs/issues/1530#issuecomment-361072194,  https://github.com/PowerPointLabs/PowerPointLabs/issues/2004

**Comment by iskandarzulkarnaien** [comment]
> 
> 
> Tested for 2016.
> 
> Sidenote: Bug still remains where it adds duplicate items to Powerpoint's clipboard pane as well as Window's clipboard history, but unsure if those can be fixed. Related: [#1530 (comment)](https://github.com/PowerPointLabs/PowerPointLabs/issues/1530#issuecomment-361072194), #2004

Alright noted, those are separate issues that I'll try to look into. Thanks!

@blewjy @YuPeiHenry @yuhongherald could one of you merge my PR? Thanks for the help!

**Pull request merged**